src/coreComponents/linearAlgebra/interfaces/hypre/HypreMatrix.cpp

```diff
@@ -162,38 +162,45 @@ void HypreMatrix::createWithGlobalSize( globalIndex const globalRows,
 void HypreMatrix::create( CRSMatrixView< real64 const, globalIndex const > const & localMatrix,
                           MPI_Comm const & comm )
 {
-  localIndex maxEntriesPerRow = 0;
-  for( localIndex i = 0; i < localMatrix.numRows(); ++i )
-  {
-    maxEntriesPerRow = std::max( maxEntriesPerRow, localMatrix.numNonZeros( i ) );
-  }
+  RAJA::ReduceMax< parallelDeviceReduce, localIndex > maxRowEntries( 0 );
+
+  forAll< parallelDevicePolicy< 32 > >( localMatrix.numRows(),
+    [localMatrix, maxRowEntries] GEOSX_HOST_DEVICE ( localIndex const row )
+    {
+      maxRowEntries.max( localMatrix.numNonZeros( row ) );
+    }
+  );
 
   createWithLocalSize( localMatrix.numRows(),
                        localMatrix.numRows(),
-                       maxEntriesPerRow,
+                       maxRowEntries.get(),
                        comm );
 
   globalIndex const rankOffset = ilower();
 
-  array1d< globalIndex > rows( localMatrix.numRows() );
-  array1d< HYPRE_Int > sizes( localMatrix.numRows() );
-  array1d< HYPRE_Int > offsets( localMatrix.numRows() );
-  localIndex const * const pSizes = localMatrix.getSizes();
-  localIndex const * const pOffsets = localMatrix.getOffsets();
+  array1d< globalIndex > rows;
+  rows.resizeWithoutInitializationOrDestruction( LvArray::MemorySpace::cuda, localMatrix.numRows() );
+  arrayView1d< globalIndex > const rowsV = rows;
 
-  // Note: this relies on the fact the localMatrix has correct sparsity on the host
-  // (e.g. it was built or synced there) because we don't move sizes/offsets to host here.
```

Comment:
The bug is that not only is `localMatrix` expected to be current (at least the sizes and offsets) on host, but it's also supposed to _be_ on host. The change from `setValues` to `zero` meant that `localMatrix` itself is on device so the pointers are device pointers.